relayer/processor/path_end.go

```diff
@@ -1,14 +1,83 @@
 package processor
 
+// PathEnd references one chain involved in a path.
+// A path is composed of two PathEnds.
 type PathEnd struct {
-	ChainID      string
-	ClientID     string
-	ConnectionID string
+	ChainID  string
+	ClientID string
+	// ConnectionIDs are tracked by pathEndRuntime in PathProcessor for known connections on this client
 
-	// TODO channel filter stuff (allowlist, denylist)
+	// can only provide one. panic if both are provided
+	AllowList []ChannelKey // only use these if provided
+	BlockList []ChannelKey // use everything except these if provided
 }
 
+// NewPathEnd constructs a PathEnd, validating initial parameters.
+func NewPathEnd(chainID string, clientID string, allowList []ChannelKey, blockList []ChannelKey) PathEnd {
+	if len(allowList) > 0 && len(blockList) > 0 {
+		panic("only one of allowlist or blocklist are allowed")
```

Comment:
I agree, once we have this plumbed up to the config level, I think it should be handled there.

Reply:
To couple this new filter strategy closer to the existing `ChannelFilter`, I changed it to `Rule` and `FilterList` parameters on the `PathEnd` instead of `AllowList`/`BlockList`. Rule can be either `allowlist` or `denylist`, same as the `ChannelFilter`, then the `FilterList` channels apply for the `Rule`.